### ISSUE TITLE
app crashes while sharing folder

### ISSUE THREAD

**Issue by nephilim75** [nextcloud/android#7989]
### Steps to reproduce
1. Enter app
2. Share a folder with NC user

### Expected behaviour
- folder should be shared

### Actual behaviour
- app crashes while sharing with following error log

<details><summary>Error log </summary>

```
******** CAUSE OF ERROR ********

java.lang.RuntimeException: Accessing result data after operation failed!
 at com.owncloud.android.lib.common.operations.RemoteOperationResult.getData(RemoteOperationResult.java:486)
 at com.owncloud.android.utils.ErrorMessageAdapter.getMessageForCreateShareOperations(ErrorMessageAdapter.java:254)
 at com.owncloud.android.utils.ErrorMessageAdapter.getMessageForResultAndOperation(ErrorMessageAdapter.java:143)
 at com.owncloud.android.utils.ErrorMessageAdapter.getErrorCauseMessage(ErrorMessageAdapter.java:79)
 at com.owncloud.android.ui.activity.FileActivity.onRemoteOperationFinish(FileActivity.java:380)
 at com.owncloud.android.ui.activity.FileDisplayActivity.onRemoteOperationFinish(FileDisplayActivity.java:1704)
 at com.owncloud.android.services.OperationsService.lambda$dispatchResultToOperationListeners$0(OperationsService.java:734)
 at com.owncloud.android.services.-$$Lambda$OperationsService$KsB1LbjHjlEBSp0h7XO24czs-OQ.run(Unknown Source:6)
 at android.os.Handler.handleCallback(Handler.java:789)
 at android.os.Handler.dispatchMessage(Handler.java:98)
 at android.os.Looper.loop(Looper.java:164)
 at android.app.ActivityThread.main(ActivityThread.java:6944)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)

******** APP INFORMATION ********
ID: com.nextcloud.client
Version: 30150090
Build flavor: gplay

******** DEVICE INFORMATION ********
Brand: samsung
Device: herolte
Model: SM-G930F
Id: R16NW
Product: heroltexx

******** FIRMWARE ********
SDK: 26
Release: 8.0.0
Incremental: G930FXXU8ETI2
```
</details>

**Comment by AndreaPro** [comment]
Yes, I am on Android 11, I have installed the app today on a fresh ROM and I have some weird issues with Sharing too. I've noticed that after the sharing if I try to referesh the nextcloud folder list it got stuck. If I kill the app and restart it the folder that I've chosen is not shared but instead is shared the folder at the upper level. Unsharing works fine. If I share the folder from a webpage no issues
